packages/fx-core/src/component/generator/templates/templateNames.ts

```diff
@@ -124,3 +128,254 @@ export const Feature2TemplateName = {
     CustomCopilotAssistantOptions.assistantsApi().id
   }`]: TemplateNames.CustomCopilotAssistantAssistantsApi,
 };
+
+export function tryGetTemplateName(inputs: Inputs): TemplateNames | undefined {
+  return inputsToTemplateName.find((item) =>
+    Object.keys(item.inputs).every((key) => item.inputs[key] === inputs[key])
+  )?.name;
+}
+
+export function getTemplateName(inputs: Inputs): TemplateNames {
+  const templateName = tryGetTemplateName(inputs);
+  if (!templateName) {
+    throw new Error("Template name not found");
+  }
+  return templateName;
+}
+
+// When multiple template name matches, only the top one will be picked.
+export const inputsToTemplateName: { inputs: { [key: string]: any }; name: TemplateNames }[] = [
```

Comment:
Can we just use a Map instead of find something in a list?

Reply:
Actually, we can't benefit from using a Map because we can't get a simple key from user inputs. We have to iterate the Map and it does not improve the performance.
But I still update to use a Map because it looks more compact than a list.